core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/R2dbcReadJournal.scala

```diff
@@ -99,6 +100,34 @@ final class R2dbcReadJournal(system: ExtendedActorSystem, config: Config, cfgPat
       .currentBySlices("currentEventsBySlices", entityType, minSlice, maxSlice, offset)
   }
 
+  /**
+   * Query events for given slices. A slice is deterministically defined based on the persistence id. The purpose is to
+   * evenly distribute all persistence ids over the slices.
+   *
+   * The consumer can keep track of its current position in the event stream by storing the `offset` and restart the
+   * query from a given `offset` after a crash/restart.
+   *
+   * The supported offset is [[TimestampOffset]] and [[Offset.noOffset]].
+   *
+   * The timestamp is based on the database `transaction_timestamp()` when the event was stored.
+   * `transaction_timestamp()` is the time when the transaction started, not when it was committed. This means that a
+   * "later" event may be visible first and when retrieving events after the previously seen timestamp may miss some
+   * events. In distributed SQL databases there can also be clock skews for the database timestamps. For that reason it
+   * will perform additional backtracking queries to catch missed events. Events from backtracking will typically be
+   * duplicates of previously emitted events. It's the responsibility of the consumer to filter duplicates and make sure
+   * that events are processed in exact sequence number order for each persistence id. Such deduplication is provided by
+   * the R2DBC Projection.
+   *
+   * Events emitted by the backtracking don't contain the event payload (`EventEnvelope.event` is null) and the consumer
+   * can load the full `EventEnvelope` with [[R2dbcReadJournal.loadEnvelope]].
```

Comment:
It's ofc undesired to use `null` for the payload. I'm thinking about introducing a new EventEnvelope which can support that better. Maybe something more specific to the new eventsBySlices query. We have anyway some mess with two EventEnvelopes, one in akka.persistence.query and another in akka.projection

Reply:
but that would be another PR since it would be a very intrusive change